samples/bot-sso-adaptivecard/nodejs/bots/botSSOAdaptiveCard.js

```diff
@@ -0,0 +1,258 @@
+
+            if (actiondata["verb"] == null) {
+                return null;
+            }
+
+            // authToken and state are absent, handle verb
+            let verb = actiondata["verb"];
+
+            var authentication = null;
+
+            // When adaptiveCard / action invoke activity from teams contains token in response to sso flow from earlier invoke.
+            if (value["authentication"] != null) {
+                authentication = value["authentication"];
+            }
+
+            if (authentication == null) {
+                // authToken is absent, handle verb
+                switch (verb) {
+                    case "initiateSSO":
+                        //when token is absent in the invoke.
+                        return this.onInitiateSSO(context);
+                }
+            }
+            else {
+                // Send adaptive card with login successful information
+                return this.createAdaptiveCardInvokeResponseAsync(authentication, context);
+            }
+        }
+
+        return null;
+    }
+
+    /**
+    * AuthToken present. Verify token in invoke payload and return AC response
+    * @param {authentication} authentication
+    */
+    async createAdaptiveCardInvokeResponseAsync(authentication) {
+        var auth_Obj = {
+            isTokenPresent: false,
+            isStatePresent: false,
+            isBasicRefresh: false
+        }
+
+        // verify token is present or not
+        if (authentication) {
+            auth_Obj.isTokenPresent = true;
+            auth_Obj.isStatePresent = true;
+            auth_Obj.isBasicRefresh = false;
+        };
+
+        // Create a Template instance from the template payload
+        var template = new ACData.Template(AdaptiveCardResponse);
+
+        // Use token or state to perform operation on behalf of user
+        var authResultData = auth_Obj.isTokenPresent ? "SSO success" : auth_Obj.isStatePresent ? "OAuth success" : "SSO/OAuth failed"
+        if (auth_Obj.isBasicRefresh) {
+            authResultData = "Refresh done";
+        }
+
+        var payloadData = {
+            authresult: authResultData
+        };
+
+        var card = template.expand(payloadData);
+        console.log("card", card);
+
+        var adaptiveCardResponse = {
+            StatusCode: 200,
+            Type: "application/vnd.microsoft.card.adaptive",
+            Value: card
+        }
+
+        await ActivityHandler.createInvokeResponse({
+            adaptiveCardResponse
+        });
+    }
+
+    /**
+   * Sends OAuth Card with signlink link
+   * @param {context} context
+   */
+    async onInitiateSSO(context) {
+        // Retrieve the OAuth Sign in Link
+        const signInLink = await context.adapter.getSignInLink(
+            context,
+            _connectionName
+        );
+
+        var tokenExchangeResponse = await this.exchangedToken(context);
+
+        var OAthCard =
+            CardFactory.oauthCard(
```

Comment:
this oauth card shoudl contain tokenExchangeResoiuce like this, which contains a GUID id
![image](https://user-images.githubusercontent.com/12997454/218966834-80c42909-7550-4e23-aec3-cc33bf167ea8.png)


How is SSO working without this? have you tested this code?

Reply:
Ok I will update accordingly..

Reply:
fixed